eth/backend.go

```diff
@@ -575,3 +578,31 @@ func (s *Ethereum) Stop() error {
 
 	return nil
 }
+
+func (s *Ethereum) HandleRequiredProtocolVersion(required params.ProtocolVersion) error {
+	var needLevel int
+	switch s.config.RollupHaltOnIncompatibleProtocolVersion {
+	case "major":
+		needLevel = 3
+	case "minor":
+		needLevel = 2
+	case "patch":
+		needLevel = 1
+	default:
+		return nil // do not consider halting if not configured to
+	}
+	haveLevel := 0
+	switch params.OPStackSupport.Compare(required) {
```

Comment:
This will do nothing on cases where the version type or build differs since `Compare` checks those before checking the major/minor/patch fields.  Halting on different build type seems better given you're on the wrong chain. The migration between version types is hard to know what to do but may be worth at least a warning.

Reply:
Ah I see, we will log a warning on these cases in the API handler. Maybe worth a comment that other cases are expected to be logged there.